src/Nox.Types/Types/DateTimeSchedule/DateTimeSchedule.cs

```diff
@@ -0,0 +1,175 @@
+    /// </summary>
+    public System.DateTime EndDate
+    {
+        get => Value.EndDate;
+        private set => Value = (StartDate: Value.StartDate, EndDate: value);
+    }
+
+    /// <summary>
+    /// Creates and validates a new instance of <see cref="DateTimeSchedule"/> from <paramref name="startDateTime"/> and <paramref name="endDateTime"/>.
+    /// </summary>
+    /// <param name="startDateTime">start date of schedule</param>
+    /// <param name="endDateTime">end date of scheduled event</param>
+    /// <returns>New <see cref="DateTimeSchedule"/> object.</returns>
+    public static DateTimeSchedule From(System.DateTime startDateTime, System.DateTime endDateTime) => From(startDateTime, endDateTime, new DateTimeScheduleTypeOptions());
+
+    /// <summary>
+    /// Creates and validates a new instance of <see cref="DateTimeSchedule"/> from <paramref name="startDateTime"/> and <paramref name="endDateTime"/>.
+    /// </summary>
+    /// <param name="startDateTime">start date of schedule</param>
+    /// <param name="endDateTime">end date of scheduled event</param>
+    /// <param name="options"><see cref="DateTimeScheduleTypeOptions"/> that contains frequency of scheduler</param>
+    /// <returns>New <see cref="DateTimeSchedule"/> object.</returns>
+    public static DateTimeSchedule From(System.DateTime startDateTime, System.DateTime endDateTime, DateTimeScheduleTypeOptions options)
+    {
+        var newObject = new DateTimeSchedule
+        {
+            Value = (startDateTime, endDateTime),
+            _dateTimeScheduleTypeOptions = options
+        };
+
+        var validationResult = newObject.Validate();
+
+        if (!validationResult.IsValid)
+        {
+            throw new TypeValidationException(validationResult.Errors);
+        }
+
+        return newObject;
+    }
+
+    /// <summary>
+    /// Validates a <see cref="DateTimeSchedule"/> object.
+    /// </summary>
+    /// <returns>True if the <see cref="DateTimeSchedule"/> value is valid, otherwise false.</returns>
+    internal override ValidationResult Validate()
+    {
+        var result = base.Validate();
+
+        // validate date by options
+        if (Value.StartDate > Value.EndDate)
+        {
+            result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox DateTimeSchedule type with Start value {Value.StartDate} and End value {Value.EndDate} as start of the time range must be the same or after the end of the time range."));
+        }
+
+        return result;
+    }
+
+    /// <summary>
+    /// Returns a string representation of the custom date and time using <see cref="CultureInfo.InvariantCulture"/>.
+    /// </summary>
+    public override string ToString()
+    {
+        return $"Start: {Value.StartDate}, End: {Value.EndDate}, Frequency: {_dateTimeScheduleTypeOptions.Frequency}, Frequency value: {_dateTimeScheduleTypeOptions.FrequencyValue}";
+    }
+
+    public DateTimeSchedule GetNextScheduledDate()
```

Comment:
I think `GetNextScheduleDate` should just return the next Date, not a new Schedule object.